File: src/main/java/no/nav/sosialhjelp/soknad/consumer/arbeidsforhold/dto/ArbeidsforholdDto.java
//package no.nav.sosialhjelp.soknad.consumer.arbeidsforhold.dto;
//
//import com.fasterxml.jackson.annotation.JsonCreator;
//import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
//import com.fasterxml.jackson.annotation.JsonProperty;
//
//import java.util.List;
//
//@JsonIgnoreProperties(ignoreUnknown = true)
//public class ArbeidsforholdDto {
//
//    private final AnsettelsesperiodeDto ansettelsesperiode;
//    private final List<ArbeidsavtaleDto> arbeidsavtaler;
//    private final String arbeidsforholdId;
//    private final OpplysningspliktigArbeidsgiverDto arbeidsgiver;
//    private final PersonDto arbeidstaker;
//    private final Long navArbeidsforholdId;
//
//    @JsonCreator
//    public ArbeidsforholdDto(
//            @JsonProperty("ansettelsesperiode") AnsettelsesperiodeDto ansettelsesperiode,
//            @JsonProperty("arbeidsavtaler") List<ArbeidsavtaleDto> arbeidsavtaler,
//            @JsonProperty("arbeidsforholdId") String arbeidsforholdId,
//            @JsonProperty("arbeidsgiver") OpplysningspliktigArbeidsgiverDto arbeidsgiver,
//            @JsonProperty("arbeidstaker") PersonDto arbeidstaker,
//            @JsonProperty("navArbeidsforholdId") Long navArbeidsforholdId) {
//        this.ansettelsesperiode = ansettelsesperiode;
//        this.arbeidsavtaler = arbeidsavtaler;
//        this.arbeidsforholdId = arbeidsforholdId;
//        this.arbeidsgiver = arbeidsgiver;
//        this.arbeidstaker = arbeidstaker;
//        this.navArbeidsforholdId = navArbeidsforholdId;
//    }
//
//    public AnsettelsesperiodeDto getAnsettelsesperiode() {
//        return ansettelsesperiode;
//    }
//
//    public List<ArbeidsavtaleDto> getArbeidsavtaler() {
//        return arbeidsavtaler;
//    }
//
//    public String getArbeidsforholdId() {
//        return arbeidsforholdId;
//    }
//
//    public OpplysningspliktigArbeidsgiverDto getArbeidsgiver() {
//        return arbeidsgiver;
//    }
//
//    public PersonDto getArbeidstaker() {
//        return arbeidstaker;
//    }
//
//    public Long getNavArbeidsforholdId() {
//        return navArbeidsforholdId;
//    }
//
//}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/client/arbeidsforhold/dto/ArbeidsforholdDto.kt
package no.nav.sosialhjelp.soknad.client.arbeidsforhold.dto

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo
import java.time.LocalDate

data class ArbeidsforholdDto(
    val ansettelsesperiode: AnsettelsesperiodeDto,
    val arbeidsavtaler: List<ArbeidsavtaleDto>,
    val arbeidsforholdId: String,
    val arbeidsgiver: OpplysningspliktigArbeidsgiverDto,
    val arbeidstaker: PersonDto,
    val navArbeidsforholdId: Long
)

data class AnsettelsesperiodeDto(
    val periode: PeriodeDto
)

data class ArbeidsavtaleDto(
    val stillingsprosent: Double
)

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
@JsonSubTypes(
    JsonSubTypes.Type(value = OrganisasjonDto::class, name = "Organisasjon"),
    JsonSubTypes.Type(value = PersonDto::class, name = "Person"),
)
sealed class OpplysningspliktigArbeidsgiverDto()

data class OrganisasjonDto(
    val organisasjonsnummer: String?,
    val type: String?
) : OpplysningspliktigArbeidsgiverDto()

data class PersonDto(
    val offentligIdent: String,
    val aktoerId: String,
    val type: String?
) : OpplysningspliktigArbeidsgiverDto()


data class PeriodeDto(
    val fom: LocalDate,
    val tom: LocalDate
)
